app/screens/home/channel_list/servers/servers_list/server_item/index.tsx

```diff
@@ -0,0 +1,257 @@
+
+import type ServersModel from '@typings/database/models/app/servers';
+import type MyChannelModel from '@typings/database/models/servers/my_channel';
+import type {Subscription} from 'rxjs';
+
+type Props = {
+    isActive: boolean;
+    server: ServersModel;
+}
+
+type BadgeValues = {
+    isUnread: boolean;
+    mentions: number;
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    active: {
+        borderColor: theme.sidebarTextActiveBorder,
+        borderWidth: 3,
+    },
+    badge: {
+        left: 18,
+        top: -5,
+    },
+    button: {
+        borderRadius: 8,
+        flex: 1,
+        height: 72,
+        justifyContent: 'center',
+        paddingHorizontal: 18,
+    },
+    container: {
+        alignItems: 'center',
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.04),
+        borderRadius: 8,
+        flexDirection: 'row',
+        height: 72,
+        marginBottom: 12,
+    },
+    details: {
+        marginLeft: 14,
+    },
+    logout: {
+        backgroundColor: theme.centerChannelBg,
+        borderRadius: 8,
+        height: 16,
+        left: 40,
+        position: 'absolute',
+        top: 11,
+        width: 16,
+    },
+    nameContainer: {
+        alignItems: 'center',
+        flexDirection: 'row',
+    },
+    name: {
+        color: theme.centerChannelColor,
+        ...typography('Body', 200, 'SemiBold'),
+    },
+    offline: {
+        opacity: 0.5,
+    },
+    row: {flexDirection: 'row'},
+    unread: {
+        top: -2,
+        left: 25,
+    },
+    url: {
+        color: changeOpacity(theme.centerChannelColor, 0.72),
+        ...typography('Body', 75, 'Regular'),
+    },
+    websocket: {
+        marginLeft: 7,
+        marginTop: 4,
+    },
+}));
+
+const ServerItem = ({isActive, server}: Props) => {
+    const intl = useIntl();
+    const theme = useTheme();
+    const [badge, setBadge] = useState<BadgeValues>({isUnread: false, mentions: 0});
+    const styles = getStyleSheet(theme);
+    const swipeable = useRef<Swipeable>();
+    const subscription = useRef<Subscription|undefined>();
+    const websocketError = server.lastActiveAt ? !WebsocketManager.isConnected(server.url) : false;
+    let displayName = server.displayName;
+    if (server.url === server.displayName) {
+        displayName = intl.formatMessage({id: 'servers.default', defaultMessage: 'Default Server'});
+    }
+
+    const unreadsSubscription = (myChannels: MyChannelModel[]) => {
+        let mentions = 0;
+        let isUnread = false;
+        for (const myChannel of myChannels) {
+            mentions += myChannel.mentionsCount;
+            isUnread = isUnread || myChannel.isUnread;
+        }
+
+        setBadge({isUnread, mentions});
+    };
+
+    const containerStyle = useMemo(() => {
+        const style = [styles.container];
+        if (isActive) {
+            style.push(styles.active);
+        }
+
+        return style;
+    }, [isActive]);
```

Comment:
This is a question for my own knowledge.  containerStyle also depends on styles.active which in turn depends on the theme.  If the theme changes, then styles.active is also changed.  But we are not including it here because this is an edge case?

Reply:
well that would only be a problem if you are using the app and change your theme through the webapp/desktop. there is no way to change your theme as this screen is being displayed from within the app.